python/cudf/cudf/core/column/numerical.py

```diff
@@ -320,13 +322,45 @@ def find_and_replace(self, to_replace, replacement, all_nan):
         replacement_col = column.as_column(
             replacement, dtype=replacement_dtype
         )
+
+        # If to_replace is a python list or scalar, check mininum viable dtype
+        if isinstance(to_replace, list) or is_scalar(to_replace):
+            to_replace_dtype = min_numeric_column_type(to_replace_col)
+        else:
+            to_replace_dtype = to_replace.dtype
+        if to_replace_dtype > self.dtype or (
+            np.issubdtype(self.dtype, np.number)
+            and np.issubdtype(to_replace_dtype, np.floating)
+        ):
+            raise TypeError(
+                "Dtype of `to_replace`: {} should be <= dtype"
+                "of the series: {}".format(to_replace_dtype, self.dtype)
+            )
+        # If replacement is a python list or scalar, check mininum viable dtype
+        if isinstance(replacement, list) or is_scalar(replacement):
+            replacement_dtype = min_numeric_column_type(replacement_col)
```

Comment:
Also doing this on device instead of host because of `min/max` performance concerns if these columns are big. Also min/max on python lists require an extra step to filter out nulls since these operations don't work with nulls.